src/fstab-generator/fstab-generator.c

```diff
@@ -451,14 +451,39 @@ static int add_mount(
                         return r;
         }
 
-        if (!(flags & NOAUTO) && !(flags & AUTOMOUNT)) {
-                r = generator_add_symlink(dest, post,
-                                          (flags & NOFAIL) ? "wants" : "requires", name);
-                if (r < 0)
-                        return r;
-        }
+        if (!(flags & AUTOMOUNT)) {
+                if (!(flags & NOAUTO)) {
+                        r = generator_add_symlink(dest, post,
+                                                  (flags & NOFAIL) ? "wants" : "requires", name);
+                        if (r < 0)
+                                return r;
+                } else {
+                        _cleanup_strv_free_ char **units = NULL;
+                        char **s;
 
-        if (flags & AUTOMOUNT) {
+                        r = fstab_extract_values(opts, "x-systemd.wanted-by", &units);
```

Comment:
hmm, thinking about this, I think it would be prettier if we'd parse this out first, and would not mangle the NOAUTO flag, but just change the `if (!(flags & NOAUTO))` check to become `if (!FLAGS_SET(flags, NOAUTO) && strv_isempty(wanted_by) && strv_isempty(required_by)` instead. I.e. mangling the parsed data early sounds confusing since our flags won't match what we read from the file. Given however that we only really need to check the combined test only once here it's prettier to just keep in memory the original parsed data, and do the check only based on that.

I hope this makes sense?

Reply:
Just to be clear, you want the behavior to be different: `noauto` should be changed to override the `required-by` and `wanted-by` directives (making them do nothing)?

Reply:
no, i am fine with the behaviour you proposed. I just want it differently implemented. instead of mangling the noauto bool/NOAUTO flag early on, just pass it on as it is, and combine with the wanted_by/required_by check where we actually need it.